js/db.js

```diff
@@ -0,0 +1,15 @@
+import * as firebase from "firebase";
+import "firebase/firestore";
+
+// Initialize Cloud Firestore through Firebase
+firebase.initializeApp({
+  apiKey: "AIzaSyCvxKWuYDgJ4r4o8JeNAOYusx0aV71_YuE",
```

Comment:
I don't think this should have been left in.

Reply:
No, thats fine - the apiKey is public info.

Reply:
Right, this config is part of the snippet provided by Firebase that is meant to be included in the HTML page. So, I'm assuming it's safe.

Reply:
Just got this email: 
>
> 
> Dear Customer,
> --
> We have detected a publicly accessible Google API key associated with the following Google Cloud Platform project:Project report-service-dev (id: report-service-dev) with API key AIzaSyCvxKWuYDgJ4r4o8JeNAOYusx0aV71_YuEThe key was found at the following URL: https://github.com/concord-consortium/portal-report/blob/95d269376073b4a0440f29d098b292457efb4785/js/db.tsWe believe that you or your organization may have inadvertently published the affected API key in public sources or on public websites (for example, credentials mistakenly uploaded to a service such as GitHub.)


![image](https://user-images.githubusercontent.com/22908/58647761-3e384c80-82d6-11e9-8b7f-8049dd3495a4.png)

Reply:
Is that from Google? It seems so. I got one from some bot, but ignored it, as it was some third-party project.

It's weird as if you go here:
https://console.firebase.google.com/u/0/project/report-service-dev/settings/general/web:NDU0ZGM1M2QtMzc2Ni00NTdjLWFlNjctODMzZGZiMzgzOWEx
There's a snippet that they recommend to past at the end of <body> tag:
<img width="631" alt="Screenshot 2019-05-31 at 12 30 47" src="https://user-images.githubusercontent.com/767857/58700334-04cc1380-83a0-11e9-93e0-c28e29ba0c2d.png">
There's a bit of polish (can't change the language of this website), but you should see something similar. If it's meant to be in the HTML page, it means it's fully public.

Also: https://stackoverflow.com/questions/37482366/is-it-safe-to-expose-firebase-apikey-to-the-public/37484053#37484053

We could possibly return apiKey together with JWT, but it seems to me like overkill. Maybe exposing that in public github repos means that it's easier for some bots to parse and abuse, but generally, it seems it's a public key.

Reply:
Noah and I fixed the issue with this yesterday.  It seems the Firebase console creates a limited api key by default (only allows access to Firebase apis) but the Google Project console creates api keys that are not limited.  We went into the Google Console and limited that key to just the Firestore api.  We might need to increase the api coverage in the future.

Reply:
Hm, today I'm getting this error when Portal Report tries to sign in to Firebase:

> "{"error":{"code":403,"message":"Requests to this API identitytoolkit method google.cloud.identitytoolkit.v1.AuthenticationService.SignInWithCustomToken are blocked.","errors":[{"message":"Requests to this API identitytoolkit method google.cloud.identitytoolkit.v1.AuthenticationService.SignInWithCustomToken are blocked.","domain":"global","reason":"forbidden"}],"status":"PERMISSION_DENIED"}}"

@dougmartin, could you try to verify that you didn't restrict access too much or revert these changes?